More stuff

ezyang opened the pull request:
More stuff

Summary: boom

Test Plan: boom

Differential Revision: https://phabricator.intern.facebook.com/D9514690

apaszke commented:
You can use a new exception type to make sure no other try blocks catch them (except for `std::exception`)

zdevito commented:
I am going to merge this now, so we have asserts and consistency checks. We fix it so that primitive ops are easier to write in a latter change if it becomes an issue.

apaszke commented on torch/csrc/jit/assert.cpp:
```diff
@@ -0,0 +1,25 @@
+
+  // Convenience for gdb; helps you avoid fumbling around
+  // to figure out how to trap on throwing a C++ exception.
+  __builtin_trap();
```
Just FYI you can stop the process in gdb at every exception if you use `catch throw`. Doesn't thi scause the process to halt immediately, in such a way that the error is never raised?

ezyang replied:
Yes, you are right! I opened a new PR to fix this. https://github.com/ezyang/pytorch/pull/9

ezyang committed:
Trivial change arf

Summary: none

Differential Revision: https://phabricator.intern.facebook.com/D9514646

ezyang committed:
More stuff

Summary: boom

Test Plan: boom

Differential Revision: https://phabricator.intern.facebook.com/D9514690
